### PR TITLE
Fixed crash when ending BT sessions

### PR DESCRIPTION
**Pull request by pawelgil**
* Made ending synchronization [polymorphic](https://en.wikipedia.org/wiki/Polymorphism_(computer_science))
* We need to sort out dependency injection...

### CODE REVIEW

File: AirCasting/RootAppView.swift [hunk 1]
```diff
@@ -41,21 +43,24 @@ struct RootAppView: View {
...
-                .environmentObject(dependancies.airBeamConnectionController)
-                .environment(\.managedObjectContext, persistenceController.viewContext)
-        }
+        let sessionStoppableFactory = SessionStoppableFactoryDefault(microphoneManager: microphoneManager,
+                                                                     measurementStreamStorage: measurementStreamStorage,
+                                                                     synchronizer: sessionSynchronizer)
+        return MainTabBarView(measurementUpdatingService: DownloadMeasurementsService(
+                                authorisationService: userAuthenticationSession,
+                                persistenceController: persistenceController,
+                                baseUrl: dependancies.urlProvider), urlProvider: dependancies.urlProvider, measurementStreamStorage: measurementStreamStorage, sessionStoppableFactory: sessionStoppableFactory, sessionSynchronizer: sessionSynchronizer, sessionContext: CreateSessionContext())
```
Comment (jankrzempek) [inline AirCasting/RootAppView.swift:52]: Maybe we can stuck this vertical to be more readable - what is your opinion about that?

**Review by jankrzempek — APPROVED** [review]
👌🏻

**Review by smialko — CHANGES_REQUESTED** [review]
App doesn't crash but finish session doesn't work either :)

**Comment by pawelgil** [comment]
> App doesn't crash but finish session doesn't work either :)

@smialko as we discussed this was not implemented at all. I resolved conflicts, please review :)

**Review by smialko — APPROVED** [review]

**Comment by smialko** [comment]
I'm merging it because I need this for the task I'm working on :)

**Pull request merged**